frontEnd/src/components/item.jsx

```diff
@@ -1,83 +1,99 @@
 //Item framework
-function Item({
-  postType,
-  postId,
-  userId,
-  isMyPost,
-  title,
-  description,
-  useState,
-  onPostChange,
-  updatePosts,
-}) {
+import React, { useRef } from "react";
+function Item({ postType, userId, isMyPost, item, onPostChange, updatePosts }) {
+  //consts needed through this element
+  const itemRef = useRef(null);
   const backendUrl = import.meta.env.VITE_BACKEND;
 
+  /*This function is called when we want to delete an item from the list*/
   const deleteItem = async (event) => {
     event.preventDefault();
     const response = await fetch(
-      `${backendUrl}users/${userId}/${postType}/${postId}`,
+      `${backendUrl}users/${userId}/${postType}/${item.id}`,
       {
         method: "DELETE",
         headers: { "Content-Type": "application/json" },
       }
     );
     onPostChange(!updatePosts);
   };
-
-  const editItem = (event) => {
+  /*function to edit an item, will grab certain values and send a put to the server */
+  const postItemEdits = async (event) => {
     event.preventDefault();
-    // const response = fetch(`${backendUrl}users/${userId}/${postType}/${postId}`, {
-    //   method: "PUT",
-    //   headers: { "Content-Type": "application/json" },
-    //   body: JSON.stringify(newData),
-    // });
+    const parentItem = itemRef.current;
+    const description = parentItem.querySelector("#description").value;
+    const title = parentItem.querySelector("#title").value;
+    const itemState = parentItem.querySelector("#useStates").value;
+
+    if (title !== "" && description !== "") {
+      const response = await fetch(
+        `${backendUrl}users/${userId}/${postType}/${item.id}`,
+        {
+          method: "PUT",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            id: item.id,
+            title: title,
+            description: description,
+            photo: "",
+            useState: itemState,
+            userId: userId,
+          }),
+        }
+      );
+      parentItem.querySelector("#title").placeholder =
+        parentItem.querySelector("#title").value;
+      parentItem.querySelector("#title").value = "";
+      parentItem.querySelector("#description").placeholder =
+        parentItem.querySelector("#description").value;
+      parentItem.querySelector("#description").value = "";
```

Comment:
i think this would be better done in a "react-y" way with states and update handlers, but it looks like it's working as implemented so should be ok. if we run into issues here in the future, would highly recommend swapping to the state approach.